manager/integration/tests/test_node.py

```diff
@@ -2239,4 +2256,282 @@ def test_node_eviction():
+        "name": "pod-data",
+        "persistentVolumeClaim": {
+            "claimName": pvc1_name
+        }
+    }]
+
+    pod1['spec']['nodeSelector'] = \
+        {"kubernetes.io/hostname": node3.name}
+
+    volume1 = create_and_check_volume(client,
+                                      volume1_name,
+                                      num_of_replicas=2,
+                                      size=volume1_size)
+
+    create_pv_for_volume(client, core_api, volume1, pv1_name)
+    create_pvc_for_volume(client, core_api, volume1, pvc1_name)
+
+    create_and_wait_pod(core_api, pod1)
+    volume1 = wait_for_volume_healthy(client, volume1_name)
+
+    volume1_replica1 = volume1.replicas[0]
+    volume1_replica2 = volume1.replicas[1]
+
+    assert volume1_replica1.mode == "RW"
+    assert volume1_replica2.mode == "RW"
+    assert volume1_replica1.running is True
+    assert volume1_replica2.running is True
+
+    write_pod_volume_random_data(core_api,
+                                 pod1_name,
+                                 volume1_data_path,
+                                 DATA_SIZE_IN_MB_2)
+
+    volume1_md5sum = get_pod_data_md5sum(core_api,
+                                         pod1_name,
+                                         volume1_data_path)
+
+    client.update(node1, allowScheduling=False, evictionRequested=True)
+    wait_for_volume_replica_count(client, volume1_name, 3)
+
+    volume1 = client.by_id_volume(volume1_name)
+
+    volume1_err_replica = None
+    for replica in volume1.replicas:
+        if replica.name == volume1_replica1.name or \
+           replica.name == volume1_replica2.name:
+            assert replica.running is True
+            assert replica.mode == "RW"
+        else:
+            volume1_err_replica = replica
+            break
+    assert volume1_err_replica is not None
+    assert volume1_err_replica.running is False
+    assert volume1_err_replica.mode == ''
+
+    client.update(node1, allowScheduling=False, evictionRequested=False)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+
+    for replica in volume1.replicas:
+        if replica.name == volume1_replica1.name or \
+           replica.name == volume1_replica2.name:
+            assert replica.running is True
+            assert replica.mode == "RW"
+
+    client.update(node1, evictionRequested=True)
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    try:
+        client.update(replica_node_soft_anti_affinity_setting,
+                      value="true")
+    except Exception as e:
+        print("\nException when update "
+              "Replica Node Level Soft Anti-Affinity setting",
+              replica_node_soft_anti_affinity_setting)
+        print(e)
+
+    wait_for_volume_replica_count(client, volume1_name, 3)
+    wait_for_volume_replica_count(client, volume1_name, 2)
```

Comment:
Why do we wait for 3 and then 2?

Reply:
Wait for replica count == 3 will indicate eviction started, then wait for replica count == 2 will indicate eviction is completed, then we can check if new scheduled replicas will land on the expected nodes.